lib_test/test.ml

```diff
@@ -39,7 +39,6 @@ let spew s filename =
 let test_roundtrip _ =
   let check_roundtrip ?printer ?msg t v =
     assert_equal ?printer ?msg v (of_string t (to_string t v))
-      ~cmp:(fun x y -> Pervasives.compare x y = 0)
```

Comment:
It looks like we need to use `compare` (rather than the default `=`) here for correct comparison of `nan` values.  Perhaps a comment would be useful.

Reply:
Good call. I reverted to the old behavior and added a comment.